examples/bin/sample_client_server.rs

```diff
@@ -0,0 +1,216 @@
+/// It uses rustls-platform-verifier to utilize your machine's certificate validation for TLS communication.
+/// Both the client and server retrieve certificates from the "CurrentUser" "my" store.
+/// Please install rustls-client.pfx and rustls-server.pfx if you want to test it.
+/// Usage: cargo run --bin test_client_server
+//  The reference for this program is https://github.com/rustls/rustls-cng/blob/dev/tests/test_client_server.rs
+
+mod client {
+
+    use std::{
+        io::{Read, Write},
+        net::{Shutdown, TcpStream},
+        sync::Arc,
+    };
+
+    use anyhow::Result;
+    use rustls::{
+        client::ResolvesClientCert, sign::CertifiedKey, ClientConfig, ClientConnection,
+        SignatureScheme, Stream,
+    };
+    use rustls_cng::{
+        signer::CngSigningKey,
+        store::{CertStore, CertStoreType},
+    };
+    use rustls_pki_types::CertificateDer;
+    use rustls_platform_verifier::Verifier;
+    use rustls_symcrypt::default_symcrypt_provider;
+
+    #[derive(Debug)]
+    pub struct ClientCertResolver(String);
+
+    fn get_chain(hex_thumbprint: &str) -> Result<(Vec<CertificateDer<'static>>, CngSigningKey)> {
+        let store = CertStore::open(CertStoreType::CurrentUser, "My")?;
+        let thumbprint = hex::decode(hex_thumbprint)?;
+        let contexts = store.find_by_sha1(thumbprint)?;
+        let context = contexts
+            .first()
+            .ok_or_else(|| anyhow::Error::msg("Client: No client cert"))?;
+        let key = context.acquire_key()?;
+        let signing_key = CngSigningKey::new(key)?;
+        let chain = context
+            .as_chain_der()?
+            .into_iter()
+            .map(Into::into)
+            .collect();
+        Ok((chain, signing_key))
+    }
+
+    impl ResolvesClientCert for ClientCertResolver {
+        fn resolve(
+            &self,
+            _acceptable_issuers: &[&[u8]],
+            sigschemes: &[SignatureScheme],
+        ) -> Option<Arc<CertifiedKey>> {
+            println!("Client sig schemes: {:#?}", sigschemes);
+            let (chain, signing_key) = get_chain(&self.0).ok()?;
+            for scheme in signing_key.supported_schemes() {
+                if sigschemes.contains(scheme) {
+                    return Some(Arc::new(CertifiedKey {
+                        cert: chain,
+                        key: Arc::new(signing_key),
+                        ocsp: None,
+                    }));
+                }
+            }
+            None
+        }
+
+        fn has_certs(&self) -> bool {
+            true
+        }
+    }
+
+    pub fn run_client(port: u16) -> Result<()> {
+        let hex_thumbprint = "a508d75eac3f646de59e406a5382ae2a40037d97";
+
+        let client_config =
+            ClientConfig::builder_with_provider(Arc::new(default_symcrypt_provider()))
+                .with_safe_default_protocol_versions()?
+                .dangerous()
```

Comment:
Might add a comment about .dangerous

Reply:
You might consider implementing an extension trait for this. 

Take a look at this bit of `rustls-platform-verifier` for inspiration:
https://github.com/rustls/rustls-platform-verifier/blob/2f240146fa935e7cbb40b1bfb3c6c1a4cf35c360/rustls-platform-verifier/src/lib.rs#L91-L111

That approach could allow a user to do this configuration task without touching `dangerous()` like:
```rust
use rustls_symcrypt::BuilderVerifierExt;
let config = ClientConfig::builder_with_provider(Arc::new(default_symcrypt_provider()))
   .with_symcrypt_verifier()
   .with_no_client_auth();
```